src/duration.rs

```diff
@@ -765,8 +968,8 @@ fn duration_print() {
         + 0.milliseconds()
         + 123.nanoseconds();
     assert_eq!(
-        format!("{}", arbitrary),
-        "14889 days 23 h 47 min 34 s 0 ms 123 ns"
+        format!("{}", arbitrary).trim(),
+        "40 years 289 days 23 h 47 min 34 s 0 ms 0 us 123 ns"
```

Comment:
Are you sure this test is correct? Can we add more `years` tests because I'm just worried we'll get the conversion wrong.

Reply:
I assumed 1 year = 365 days, as this is only used for Display. It's easy to rollback the `years` part of the decomposition if you're not comfortable with it, I just thought I'd roll with it as it was basically free in the new Display impl. We could indeed add more `years` tests though

Reply:
I think that the IETF or someone defines a year as 365.25 days (since one century is 36525 days). But the problem is that it's an approximate definition, and that's why lots of implementations don't allow for year based computations: it's confusing.

Reply:
Understood, it makes sense, I will remove this part of the decomposition then